### DISCUSSION TITLE
Method of path module routing?

### DISCUSSION THREAD

**Discussion by jackHedaya** [rwf2/Rocket#2004] [Questions] [answered]
Hey!

Is there a way to group routes together on certain dynamic paths using Rocket? For example,

`/client/<client>/`

- `/signup`
- `/signin`
- `/forgot_password`
- `/2fa`

`business/<business>`

- `/say_hello`
- `/say_world`

I know all these routes could be declared by themselves but it would be great to group by module or something. Thanks!

**Comment by somehowchris** [comment, marked as answer]
@jackHedaya Totally with you on that. It would be amazing to have these parameters in that „base path“.

What you currently could do:
`/client/`
 - `/<client>/signup`
 - `/<client>/signin`
 - `/<client>/forgot_password`
 - `/<client>/2fa`

`/business/`
 - `/<business>/say_hello`
 - `/<business>/say_world`

**Reply by jackHedaya** [reply]
Gotcha, I guess it's not such a big deal to keep rewriting the base path. Thanks for your response!

**Comment by zertyz** [comment]
To accomplish this, I specify the base path when mounting routes -- you can concatenate several mount calls
... and each module declares it's routes as being in the root path.

Example:

main:
```
    rocket::build()
        .mount(api::BASE_PATH, routes![
            api::hello,
        ])
        .mount(backend::BASE_PATH, routes![
            backend::info,
        ])
        .launch().await
```

api module:
```
pub const BASE_PATH: &str = "/api";
#[get("/hello")]
pub fn hello() -> &'static str {
    "Hello, world!"
}
```

backend module:
```
pub const BASE_PATH: &str = "/backend";
#[get("/info")]
pub fn info() -> &'static str {
    "Backend written in Rust!"
}
```

$ curl localhost:8000/api/hello
Hello, world!
$ curl localhost:8000/backend/info
Backend written in Rust!
